testsuite/classlibrary/TestServerClientID.sc

```diff
@@ -36,16 +36,17 @@ TestServer_clientID : UnitTest {
 
 	test_userSpecifiedClientID {
 		var options = ServerOptions.new;
-		var s = Server(\testserv, NetAddr("localhost", 57111), options, 1);
-		this.assert(s == nil, "Making a server with invalid clientID should be blocked.");
+		var s = Server(\test_userSpecifiedClientID, nil, options, 1);
+		this.assert(s.clientID.isNil, "Making a server with invalid clientID should return a server with clientID nil.");
 		options.maxLogins_(8);
-		s = Server(\testserv, NetAddr("localhost", 57111), options, 7);
+		s.remove;
```

Comment:
I'm confused about your use of `Server.remove` in these Unit Tests. Can you explain what it does and why it is necessary?

Reply:
when you create a server, it is (implicitly) added to the global list of servers. `remove` removes it again, it's not needed anymore, so it can be garbage collected.

Reply:
I wonder why `options.maxLogins_(8);` precedes this.

Reply:
ok, I see now that he is creating Server objects in his tests and removing them at the end.

Reply:
I don't think `options.maxLogins_(8)` needs to precede `s.remove`. It might be clearer if those lines where inverted.

Reply:
yes, you are right, this can go. It was left over from a following test that became obsolete.

Reply:
just going over this again - we need to set maxLogins here because it sets the limit
for allowed clientIDs. I added explaining comments in the code here:
```
	test_userSpecifiedClientID {
		var options = ServerOptions.new;
		// options.maxLogins is 1 by default, so only clientID 0 is allowed:
		var s = Server(\testserv, NetAddr("localhost", 57111), options, 1);
		this.assert(s == nil, "Making a server with invalid clientID should be blocked.");

		// setting maxLogins high enough to let clientID 7 just pass:
		options.maxLogins_(8);
		s = Server(\testserv, NetAddr("localhost", 57111), options, 7);
		this.assert(s.clientID == 7, "Making a server with valid nonzero clientID should work.");
		s.remove;
	}
```